dev/com.ibm.ws.security.jwtsso/src/com/ibm/ws/security/jwtsso/internal/JwtSsoBuilderComponent.java

```diff
@@ -38,329 +44,361 @@
+        }
+
+        if (props == null || props.isEmpty()) {
+            return;
+        }
+        setCookiePathToWebAppContextPath = (Boolean) props
+                .get(JwtSsoConstants.CFG_KEY_SETCOOKIEPATHTOWEBAPPCONTEXTPATH);
+        includeLtpaCookie = (Boolean) props.get(JwtSsoConstants.CFG_KEY_INCLUDELTPACOOKIE);
+        fallbackToLtpa = (Boolean) props.get(JwtSsoConstants.CFG_KEY_FALLBACKTOLTPA);
+        cookieSecureFlag = (Boolean) props.get(JwtSsoConstants.CFG_KEY_COOKIESECUREFLAG);
+        jwtBuilderRef = JwtUtils.trimIt((String) props.get(JwtSsoConstants.CFG_KEY_JWTBUILDERREF));
+        isDefaultBuilder = false;
+        if (jwtBuilderRef == null) {
+            setJwtSsoBuilderDefaults();
+            isDefaultBuilder = true;
+        }
+        jwtConsumerRef = JwtUtils.trimIt((String) props.get(JwtSsoConstants.CFG_KEY_JWTCONSUMERREF));
+        cookieName = JwtUtils.trimIt((String) props.get(JwtSsoConstants.CFG_KEY_COOKIENAME));
+        if (tc.isEntryEnabled()) {
+            Tr.exit(tc, "process");
+        }
+    }
+
+    private void setJwtSsoBuilderDefaults() {
+        jwtBuilderRef = getId();
+        signatureAlgorithm = "RS256";
+        if (tc.isDebugEnabled()) {
+            Tr.debug(tc, "builder id = ", jwtBuilderRef);
+        }
+        issuerUtil = new IssuerUtil();
+
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String getId() {
+        // TODO Auto-generated method stub
+        return getUniqueId();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public List<String> getAudiences() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String getSignatureAlgorithm() {
+        // TODO Auto-generated method stub
+        return signatureAlgorithm;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String getSharedKey() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String getTrustStoreRef() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String getTrustedAlias() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    public String getUniqueId() {
+        // TODO Auto-generated method stub
+        return uniqueId;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String getIssuerUrl() {
+        return getResolvedHostAndPortUrl();
+
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean isJwkEnabled() {
+        // TODO Auto-generated method stub
+        return false;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public long getValidTime() {
+        long result = 0;
+        if (isDefaultBuilder) {
+            result = 2; // hour, for now
+        } else {
+            boolean haveNull = jwtServiceMapRef.getReference(jwtBuilderRef) == null ||
```

Comment:
Could extract the `jwtServiceMapRef.getReference(jwtBuilderRef)` segment to a variable so it doesn't have to be called multiple times.